### DISCUSSION TITLE
Example of Bucket4J Leaky bucket implementation

### DISCUSSION THREAD

**Discussion by piyush-harness** [vladimir-bukhtoyarov/bucket4j#245] [General]
Hi,

I was looking for an example for implementing leaky-bukcet rate limiter with bucket4j and redis, Could not find any after hours of searching. Can someone help with with a simple example?

**Comment by vladimir-bukhtoyarov** [comment]
Hello @piyush-harness 

Bucket4j implements token-bucket. Only [Redisson](https://github.com/redisson/redisson) is supported as client for Rediss. You can find an example of [RedissonBasedProxyManager](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-redis/src/main/java/io/github/bucket4j/redis/redisson/cas/RedissonBasedProxyManager.java) instantiation in [this test](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-redis/src/test/java/io/github/bucket4j/redis/redisson/cas/RedissonBasedProxyManagerTest.java).
```xml
<dependency>
            <groupId>com.github.vladimir-bukhtoyarov</groupId>
            <artifactId>bucket4j-redis</artifactId>
            <version>7.3.0</version>
</dependency>
```

API of RedissonBasedProxyManager is absolutely the same with other back-ends, examples of distributed usage can be found here https://bucket4j.com/7.3.0/toc.html#distributed-facilities

**Reply by piyush-harness** [reply]
Thanks @vladimir-bukhtoyarov for quick response , However the Github page mentions that it's implemented on top of token bukcet/leaky bucket.

**Reply by vladimir-bukhtoyarov** [reply]
Description has been fixed
